File: Day3/C++/hammingDistance.cpp
/*
* @author : imkaka
* @date : 24/12/2018
*/

#include<iostream>
#include<string>

using namespace std;

int main(){

    string str1, str2;

    cin >> str1 >> str2;

    int dis = 0;
    if(str1.size() != str2.size())
        cout << "Not Valid Strings for Hamming Distance Problem!!" << endl;
    else{

        for(int i=0; i < str1.size(); ++i){

            if(str1[i] != str2[i]) dis++;

        }

        cout << "Hamming Distance b/w " << str1 << " and " << str2 << " is " << dis << "." << endl;
    }

    return 0;
}

File: Day3/JavaScript/hammingDistance.js
/*
 * @author Nomit Pahuja
 * date: 09/02/2018
 */

function hamming (str1, str2) {
    let distance = 0;

    //Check the length of strings
    if (str1.length !== str2.length) {
        console.log("Please enter strings of same length.");
        return -1;
    }

    //Find the hamming distance
    for (let i = 0; i < str1.length ; i++) {
        if(str1[i] !== str2[i]) {
            distance++;
        }
    }

    //Print the hamming distance
    console.log(`The hamming distance between the strings : ${str1} and ${str2} is ${distance}`);
    return distance;
}

hamming("strawberry","avocado");
hamming("john","taylor");
hamming("issac","rahul");